src/lib/crypto/kimchi_backend/gadgets/foreign_field.ml

```diff
@@ -111,17 +108,22 @@ let field_standard_limbs_to_bignum_bigint_standard_limbs (type f)
   , Common.field_to_bignum_bigint (module Circuit) l2 )
 
 (* Convert field standard_limbs to Bignum_bigint.t *)
-let field_standard_limbs_to_bignum_bigint (type f)
+let field_const_standard_limbs_to_bignum_bigint (type f)
     (module Circuit : Snarky_backendless.Snark_intf.Run with type field = f)
     (field_limbs : f standard_limbs) : Bignum_bigint.t =
   let l0, l1, l2 =
-    field_standard_limbs_to_bignum_bigint_standard_limbs
+    field_const_standard_limbs_to_bignum_bigint_standard_limbs
       (module Circuit)
       field_limbs
   in
   Bignum_bigint.(l0 + (Common.two_to_limb * l1) + (two_to_2limb * l2))
 
 (* Foreign field element interface *)
+(* TODO: It would be better if this were created with functor that
+ *       takes are arguments the native field and the foreign field modulus.
+ *       Then when creating foreign field elements it could check that
+ *       they are valid (less than the foreign field modulus).
+ *)
```

Comment:
But then we wouldn't be able to store known larger values such as 2^264, or in the keccak/ecdsa interface, the ffadd for `z + 0 = 1 * f + z'` when the given keccak output is 256-bits but larger than `f`.

Reply:
Exactly, however, since doing that is a special case (low-level thing we use under the hood) we could create a flag for disabling this check.  This todo is about prioritising user safety (a safe API) and making any special tricks we do the exception to the rule.

Anyway, I'm still not 100% sure we'd want to follow this TODO and have `Foreign_field` check the modulus.  Just an idea.

Reply:
Good question